pkg/scheduler/algorithm/priorities/resource_limits.go

```diff
@@ -42,7 +42,11 @@ func ResourceLimitsPriorityMap(pod *v1.Pod, meta interface{}, nodeInfo *schedule
 	allocatableResources := nodeInfo.AllocatableResource()
 
 	// compute pod limits
-	podLimits := getResourceLimits(pod)
```

Comment:
Please move `getResourceLimits` to metadata as it doesn't belong here anymore

Reply:
I add fallback logic when metadata is failed to convert to `priorityMetadata` which has the consistent logic with `resource_allocation.go`.

```go
	var requested schedulernodeinfo.Resource
	if priorityMeta, ok := meta.(*priorityMetadata); ok {
		requested = *priorityMeta.nonZeroRequest
	} else {
		// We couldn't parse metadata - fallback to computing it.
		requested = *getNonZeroRequests(pod)
	}
```

Reply:
sgtm